run_batch_general.py

```diff
@@ -1,213 +1,516 @@
-                        templateMessage_analysis = {'Metadata': 'Metadata_Plate='+toanalyze+',Metadata_Well='+eachrow+well_format %eachcol+',Metadata_Site='+str(eachsite),
-                                        'pipeline': posixpath.join(batchpath,batchpipenameanalysis),
-                                        'output': analysisoutpath,
-                                        'output_structure':anlysisoutputstructure,
-                                        'input':inputpath,
-                                        'data_file': posixpath.join(batchpath,batchpipenameanalysis)
-                                        }
-
-                    analysisqueue.scheduleBatch(templateMessage_analysis)
-
-    print('Analysis job submitted. Check your queue')
-
-#MakeZprojJobs(batch=False)    
-#MakeIllumJobs(batch=False)
-#MakeQCJobs(batch=False)
-#MakeQCJobs_persite(batch=False)
-#MakeAssayDevJobs(batch=False)
-#MakeAnalysisJobs(batch=False)
+            if not batchfile:
+                batchfile = "Batch_data_analysis.h5"
+            for plate in platelist:
+                for eachrow in rows:
+                    for eachcol in columns:
+                        for eachsite in sites:
+                            templateMessage_analysis = {
+                                "Metadata": f"Metadata_Plate={plate},Metadata_Well={eachrow}{eachcol:{well_format}},Metadata_Site={str(eachsite)}",
+                                "pipeline": posixpath.join(batchpath, batchfile),
+                                "output": outpath,
+                                "output_structure": outputstructure,
+                                "input": inputpath,
+                                "data_file": posixpath.join(batchpath, batchfile),
+                            }
+                            analysisqueue.scheduleBatch(templateMessage_analysis)
+
+        print("Analysis job submitted. Check your queue")
+
+    else:
+        print(f"Step {step} not supported.")
+
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser(
+        description="Make jobs for Distributed-CellProfiler"
+    )
+    parser.add_argument(
+        "step",
+        help="Step to make jobs for. Supported steps are zproj, illum, qc, qc_persite, assaydev, analysis",
+    )
+    parser.add_argument(
+        "path_style",
+        help="Style of input/output path. default or cpg (for Cell Painting Gallery structure).",
+    )
+    parser.add_argument("identifier", help="Project identifier")
+    parser.add_argument("batch", help="Name of batch")
+    parser.add_argument("platelist", type=lambda s: list(s.split(",")), help="List of plates to process")
+    parser.add_argument(
+        "--source",
+        dest="source",
+        default="source_4",
+        help="For Cell Painting Gallery, what is the source (nesting under project identifier).",
+    )
+    parser.add_argument(
+        "--plate-format",
+        dest="plate_format",
+        default=384,
```

Comment:
Make sure that here and elsewhere (ie rows and columns), the default here matches the function default. Otherwise, if the user DOESN'T pass --plate-format, it will default to 384, so even if they passed rows, it would overwrite them with the 384 plate format.